### ISSUE TITLE
Unable to build the UbuntuMinimal image

### ISSUE THREAD

**Issue by pmatthews05** [actions/runner-images#10028]
### Description

Trying to build the UbuntuMinimal from the GenerateResourcesAndImage.ps1 file. Running this from a Windows 11, using PowerShell 7.

```ps
GenerateResourcesAndImage -SubscriptionId:$subscriptionId `
                          -ResourceGroupName:$RGName `
                          -AzureLocation:uksouth `
                          -ImageType:UbuntuMinimal
```

Within a few seconds the following error message appears.

```
Latest commit: Date: Mon Jun 10 08:45:22 2024 +0200; Commit: 2e3a8e1ccfd026ace20f6281152ca0d81a6eafd8 - [windows] Modify TortoiseSVN installation (#10010); Author: Shamil Mubarakshin <127750046+shamil-mubarakshin@users.noreply.github.com>.
Downloading packer plugins...
Validating packer template...
Error: Failed preparing provisioner-block "shell" ""

  on C:\Code\github\runner-images\images\ubuntu\templates\ubuntu-minimal.pkr.hcl line 185:
  (source code not available)

C:\Code\github\runner-images\images\ubuntu\templates\ubuntu-minimal.pkr.hcl:186,46-67:
Unsupported attribute; This object does not have an attribute named
"helper_script_folder".


Exception: C:\Code\github\runner-images\helpers\GenerateResourcesAndImage.ps1:238:9
Line |
 238 |          throw "Packer template validation failed."
     |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Packer template validation failed.
```

### Platforms affected

- [ ] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] Ubuntu 24.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [ ] macOS 14 Arm64
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

Running this directly from today's "Main" branch.

### Is it regression?

Yes. Last time I ran this was back in January 2024. I haven't attempted since.

### Expected behavior

To build the Ubuntu Minimal image

### Actual behavior

It fails

### Repro steps

Follow the https://github.com/actions/runner-images/blob/main/docs/create-image-and-azure-resources.md guide.

**Comment by pmatthews05** [comment]
I've taken a quick look, I've worked out the problem. It's referencing a var instead of a local.

https://github.com/actions/runner-images/blob/main/images/ubuntu/templates/ubuntu-minimal.pkr.hcl#L186 should be 

`environment_vars = ["HELPER_SCRIPTS=${local.helper_script_folder}","DEBIAN_FRONTEND=noninteractive"]`

currently it says 
`environment_vars = ["HELPER_SCRIPTS=${var.helper_script_folder}","DEBIAN_FRONTEND=noninteractive"]`

**Comment by ijunaidm** [comment]
@pmatthews05 - Thank you for bringing this issue to us, we are looking into this issue, we will update you.

**Comment by ijunaidm** [comment]
@pmatthews05 - Can you please confirm if the issue is fixed after our changes. If you are facing any related issue, kindly let us know .

**Comment by veronicagt094** [comment]
Hello everyone, when making the change in the ubuntu-minimal.pkr.hcl file I get the following error:


2024/06/12 14:38:01 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: 2024/06/12 14:38:01 [ERROR] Remote command exited with '1': sudo sh -c 'DEBIAN_FRONTEND='noninteractive' HELPER_SCRIPTS='/imagegeneration/helpers' INSTALLER_SCRIPT_FOLDER='/imagegeneration/installers' PACKER_BUILDER_TYPE='azure-arm' PACKER_BUILD_NAME='build_image'  /tmp/script_909.sh'

**Comment by veronicagt094** [comment]
> @veronicagt094- ¿Puede enviarnos los registros detallados con errores?

2024/06/12 14:37:59 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: 2024/06/12 14:37:59 [DEBUG] Opening new ssh session
2024/06/12 14:37:59 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: 2024/06/12 14:37:59 [DEBUG] Starting remote scp process:  scp -vt /tmp
2024/06/12 14:37:59 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: 2024/06/12 14:37:59 [DEBUG] Started SCP session, beginning transfers...
2024/06/12 14:37:59 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: 2024/06/12 14:37:59 [DEBUG] Copying input data into temporary file so we can read the length
2024/06/12 14:37:59 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: 2024/06/12 14:37:59 [DEBUG] scp: Uploading script_909.sh: perms=C0644 size=615
2024/06/12 14:37:59 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: 2024/06/12 14:37:59 [DEBUG] SCP session complete, closing stdin pipe.
2024/06/12 14:37:59 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: 2024/06/12 14:37:59 [DEBUG] Waiting for SSH session to complete.
2024/06/12 14:37:59 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: 2024/06/12 14:37:59 [DEBUG] scp stderr (length 69): Sink: C0644 615 script_909.sh
2024/06/12 14:37:59 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: scp: debug1: fd 0 clearing O_NONBLOCK
2024/06/12 14:37:59 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: 2024/06/12 14:37:59 [DEBUG] Opening new ssh session
2024/06/12 14:37:59 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: 2024/06/12 14:37:59 [DEBUG] starting remote command: chmod 0755 /tmp/script_909.sh
2024/06/12 14:37:59 [INFO] RPC client: Communicator ended with: 0
2024/06/12 14:37:59 [INFO] RPC endpoint: Communicator ended with: 0
2024/06/12 14:37:59 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: 2024/06/12 14:37:59 [INFO] RPC endpoint: Communicator ended with: 0
2024/06/12 14:37:59 packer-provisioner-shell plugin: [INFO] RPC client: Communicator ended with: 0
2024/06/12 14:37:59 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: 2024/06/12 14:37:59 [DEBUG] Opening new ssh session
2024/06/12 14:37:59 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: 2024/06/12 14:37:59 [DEBUG] starting remote command: sudo sh -c 'DEBIAN_FRONTEND='noninteractive' HELPER_SCRIPTS='/imagegeneration/helpers' INSTALLER_SCRIPT_FOLDER='/imagegeneration/installers' PACKER_BUILDER_TYPE='azure-arm' PACKER_BUILD_NAME='build_image'  /tmp/script_909.sh'
==> azure-arm.build_image: Downloading package from https://releases.hashicorp.com/packer/1.11.0/packer_1.11.0_linux_amd64.zip to /tmp/packer_1.11.0_linux_amd64.zip...
==> azure-arm.build_image: Package downloaded in 1 seconds
==> azure-arm.build_image: replace /usr/local/bin/LICENSE.txt? [y]es, [n]o, [A]ll, [N]one, [r]ename:  NULL
==> azure-arm.build_image: (EOF or read error, treating as "[N]one" ...)
##[debug]Agent environment resources - Disk: / Available 29868.00 MB out of 85160.00 MB, Memory: Used 1322.00 MB out of 7940.00 MB, CPU: Usage 0.58%
2024/06/12 14:38:01 [INFO] 0 bytes written for 'stdout'
2024/06/12 14:38:01 [INFO] 299 bytes written for 'stderr'
2024/06/12 14:38:01 [INFO] RPC client: Communicator ended with: 1
2024/06/12 14:38:01 [INFO] RPC endpoint: Communicator ended with: 1
2024/06/12 14:38:01 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: 2024/06/12 14:38:01 [ERROR] Remote command exited with '1': sudo sh -c 'DEBIAN_FRONTEND='noninteractive' HELPER_SCRIPTS='/imagegeneration/helpers' INSTALLER_SCRIPT_FOLDER='/imagegeneration/installers' PACKER_BUILDER_TYPE='azure-arm' PACKER_BUILD_NAME='build_image'  /tmp/script_909.sh'
2024/06/12 14:38:01 packer-plugin-azure_v1.4.5_x5.0_linux_amd64 plugin: 2024/06/12 14:38:01 [INFO] RPC endpoint: Communicator ended with: 1
2024/06/12 14:38:01 packer-provisioner-shell plugin: [INFO] 0 bytes written for 'stdout'
2024/06/12 14:38:01 packer-provisioner-shell plugin: [INFO] 299 bytes written for 'stderr'
2024/06/12 14:38:01 packer-provisioner-shell plugin: [INFO] RPC client: Communicator ended with: 1
2024/06/12 14:38:01 [INFO] (telemetry) ending shell

**Comment by pmatthews05** [comment]
I can confirm I can build the image again. Many thanks.
